contracts/src/PBHEntryPointImplV1.sol

```diff
@@ -156,49 +121,40 @@ contract PBHVerifierImplV1 is WorldIDImpl {
+    /// Execute a batch of UserOperation with Aggregators
+    /// @param opsPerAggregator - The operations to execute, grouped by aggregator (or address(0) for no-aggregator accounts).
+    /// @param beneficiary      - The address to receive the fees.
+    function handleAggregatedOps(
+        IEntryPoint.UserOpsPerAggregator[] calldata opsPerAggregator,
+        address payable beneficiary
     ) external virtual onlyProxy onlyInitialized {
-        // First, we make sure this nullifier has not been used before.
-        if (nullifierHashes[nullifierHash]) revert InvalidNullifier();
-
-        // Verify the external nullifier
-        PBHExternalNullifier.verify(pbhExternalNullifier, numPbhPerMonth);
-
-        // If worldId address is set, proceed with on chain verification,
-        // otherwise assume verification has been done off chain by the builder.
-        if (address(worldId) != address(0)) {
-            // We now generate the signal hash from the sender, nonce, and calldata
-            uint256 signalHash = abi.encodePacked(sender, nonce, callData).hashToField();
-
-            // We now verify the provided proof is valid and the user is verified by World ID
-            worldId.verifyProof(root, GROUP_ID, signalHash, nullifierHash, pbhExternalNullifier, proof);
+        bytes32 hashedOps = keccak256(abi.encode(opsPerAggregator));
+        assembly ("memory-safe") {
+            tstore(_HASHED_OPS_SLOT, hashedOps)
         }
 
-        // We now record the user has done this, so they can't do it again (proof of uniqueness)
-        nullifierHashes[nullifierHash] = true;
+        for (uint256 i = 0; i < opsPerAggregator.length; ++i) {
+            PBHPayload[] memory pbhPayloads = abi.decode(opsPerAggregator[i].signature, (PBHPayload[]));
+            for (uint256 j = 0; j < pbhPayloads.length; ++j) {
+                verifyPbhProof(
+                    opsPerAggregator[i].userOps[j].sender,
+                    opsPerAggregator[i].userOps[j].nonce,
+                    opsPerAggregator[i].userOps[j].callData,
+                    pbhPayloads[j]
+                );
+            }
+        }
 
-        emit PBH(root, sender, nonce, callData, pbhExternalNullifier, nullifierHash, proof);
+        _entryPoint.handleAggregatedOps(opsPerAggregator, beneficiary);
     }
 
-    /// @notice Sets a new value for numPbhPerMonth.
-    /// @param _newValue The new value set.
-    /// @dev Can only be set by the owner.
-    function setNumPbhPerMonth(uint8 _newValue) external virtual onlyOwner onlyProxy onlyInitialized {
-        numPbhPerMonth = _newValue;
+    /// @notice Validates the hashed operations is the same as the hash transiently stored.
+    /// @param hashedOps The hashed operations to validate.
+    function validateSignaturesCallback(bytes32 hashedOps) external view virtual onlyProxy onlyInitialized {
+        assembly ("memory-safe") {
+            if iszero(eq(tload(_HASHED_OPS_SLOT), hashedOps)) { revert(0, 0) }
```

Comment:
Can we use `transient` keyword instead of assembly?

https://docs.soliditylang.org/en/v0.8.28/contracts.html#transient-storage

Reply:
seems that VSCode's solidity support doesn't recognize it but seems to compile

Reply:
This doesn't compile on my end for some reason on 0.8.28 solc 😅 , evm version is `cancun` in the `foundry.toml` as well.

Reply:
odd, let's leave it like that